src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data_clear_experience_configs.py

```diff
@@ -0,0 +1,565 @@
+
+    delete_experiences(bind)
+    delete_experience_configs(bind)
+
+
+def determine_needed_experience_configs(bind):
+    """
+    Our data model changes necessitate the creation of new Experience Configs. Now that "where" a Notice is served
+    is being moved from the "Notice" constructs to "Experiences", existing Experience Configs are not sufficient
+    to capture the full range of configuration on current Notices.
+    """
+
+    notice_query = """select id, name, regions, displayed_in_overlay, displayed_in_privacy_center, notice_key from privacynotice;"""
+    experience_config_query = """SELECT id, component, disabled, accept_button_label, banner_enabled, description, privacy_preferences_link_label, privacy_policy_link_label, privacy_policy_url, save_button_label, title, banner_description, banner_title, reject_button_label, acknowledge_button_label from privacyexperienceconfig;"""
+
+    existing_notices: ResultProxy = bind.execute(text(notice_query))
+    for existing_notice in existing_notices:
+        notice_key = existing_notice["notice_key"]
+        component_mapping = NOTICES_TO_EXPERIENCE_CONFIG.get(notice_key)
+        if component_mapping:
+            if existing_notice["displayed_in_overlay"]:
+                experience_config = component_mapping.get(ComponentType.Overlay)
+                reconciled_experience_config_map[experience_config.name][
+                    "regions"
+                ].update(existing_notice["regions"])
+                reconciled_experience_config_map[experience_config.name][
+                    "privacy_notices"
+                ].add(existing_notice["id"])
+                reconciled_experience_config_map[experience_config.name][
+                    "needs_migration"
+                ] = True
+
+            if existing_notice["displayed_in_privacy_center"]:
+                experience_config = component_mapping.get(ComponentType.PrivacyCenter)
+                reconciled_experience_config_map[experience_config.name][
+                    "regions"
+                ].update(existing_notice["regions"])
+                reconciled_experience_config_map[experience_config.name][
+                    "privacy_notices"
+                ].add(existing_notice["id"])
+                reconciled_experience_config_map[experience_config.name][
+                    "needs_migration"
+                ] = True
+
+    existing_experience_configs: ResultProxy = bind.execute(
+        text(experience_config_query)
+    )
+    has_existing_experience_configs = False
+    for eec in existing_experience_configs:
+        has_existing_experience_configs = True
+        component = ComponentType(eec["component"])
+        configs = []
+        if eea_config_id := EEA_MAPPING.get(component):
+            configs.append(reconciled_experience_config_map.get(eea_config_id.name))
+        if us_config_id := US_MAPPING.get(component):
+            configs.append(reconciled_experience_config_map.get(us_config_id.name))
```

Comment:
This is particularly clever, letting original experience config text changes be copied over to potentially multiple new experience configs.  I'd add more code comments here -

Also the decision to not ship with these GPP experiences was helpful!

Reply:
yup, that was a lifesaver 😅 

will add some more code comments for sure 👍